mypy/semanal.py

```diff
@@ -7344,13 +7346,16 @@ def is_unannotated_any(typ_: Type) -> bool:
             defn.type.ret_type = NoneType()
         if self.options.infer_function_types:
             for i, arg in enumerate(defn.arguments):
+                ret = None
                 if is_unannotated_any(defn.type.arg_types[i]):
                     if arg.variable.is_inferred and arg.initializer:
                         ret = self.analyze_simple_literal_type(
                             arg.initializer, False, do_inner=True
                         )
-                        if ret:
-                            defn.type.arg_types[i] = ret
+                    if ret:
+                        defn.type.arg_types[i] = ret
+                    if all(char == "_" for char in arg.variable.name):
```

Comment:
since this check is duplicated can it be abstracted into a function

Reply:
yeah